Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,315 @@ public static class BalanceStats
+            }
+            return peopleWithSameLowBalance + minResult;
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if(IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal max = decimal.MinValue;
+            string result = "";
+            string richestPeople = "";
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+
+                string value = auxBalances[^1..][0];
+                decimal.TryParse(value, out decimal currentBalance);
+
+                if (max < currentBalance)
+                {
+                    max = currentBalance;
+                    richestPeople = auxBalances[0];
+                    moreThanOne = false;
+                    result = $" is the richest person. {max:C0}.";
+                }
+                else if(max.Equals(currentBalance))
+                {
+                    richestPeople += $", {auxBalances[0]}";
+                    result = $" are the richest people. {max:C0}.";
+                    moreThanOne = true;
+                }
+            }
+
+            if (moreThanOne)
+            {
+                richestPeople = ReplaceLastCommaWithAnd(richestPeople);
+            }
+
+            return richestPeople + result;
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal min = decimal.MaxValue;
+            string result = "";
+            string poorPeople = "";
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+                string value = auxBalances[^1..][0];
+                decimal.TryParse(value, out decimal currentBalance);
+                if (min > currentBalance)
+                {
+                    min = currentBalance;
+                    poorPeople = auxBalances[0];
+                    result = $" has the least money. {min:C0}.";
+                    moreThanOne = false;
+                }
+                else if(min.Equals(currentBalance))
+                {
+                    poorPeople += $", {auxBalances[0]}";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5069037)